quaternions.scad

```diff
@@ -73,6 +104,23 @@ function QuatXYZ(a=[0,0,0]) =
     Q_Mul(qz, Q_Mul(qy, qx));
 
 
+// Function: Q_From_to()
+// Usage:
+//    q = Q_From_to(v1, v2);
+// Description: 
+//   Returns the normalized quaternion that rotates the non zero 3D vector v1 
+//   to the non zero 3D vector v2.
+function Q_From_to(v1, v2) =
+  assert( is_vector(v1,3) && is_vector(v2,3) 
+          && ! approx(norm(v1),0) && ! approx(norm(v2),0)

```

Comment:
The `!approx(norm(v),0)` idiom seems to occur frequently.  Do you think we should have a `null_vec(x)` function?  `vzero(x)`?  `is_zero(x)`?  `is_null(x)`?

Reply:
It would be preferable but in order to avoid the complement operator (!), something like `non_zero(x)` should be defined as well. Maybe `is_zero` could be defined not only for checking scalars but also of more complex lists. (Note that `approx([0,0],0)` is false.) Something like this:

```
function is_zero(a,eps=EPSILON) =
  (is_num(a) && abs(a)<eps) ||
  (is_list(a) && len(a)>0 && ([]==[for (i=idx(a)) if(! is_zero(a[i] , eps)) 1]));
```
There is room for different criteria but this definition will return true for all the following calls:

```
echo(is_zero([0,0]));
echo(is_zero([0,[0]]));
echo(is_zero([1e-12,0]));
echo(is_zero([0,[0,[1e-12]]]));
echo(! is_zero(1));
echo(! is_zero([1,2,3]));
echo(! is_zero([]));
echo(! is_zero([0,[]]));
echo(! is_zero([1,2,["a"]]));
echo(! is_zero("a"));
echo(! is_zero([0,"a"]));
echo(! is_zero([0,1]));
echo(! is_zero([undef,1]));
echo(! is_zero([1,2,undef]));
echo(! is_zero([0,[0,[undef]]]));
echo(! is_zero([0,[0,[]]]));
```

That is: it is true if and only if all entries of the checking value are numerically near zero. Void list are not zero.

Reply:
Deferring this into it's own issue.